homeassistant/components/binary_sensor/tapsaff.py

```diff
@@ -0,0 +1,85 @@
+"""
+Support for Taps Aff binary sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.tapsaff/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor \
+    import (BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['tapsaff==0.1.0']
+
+CONF_LOCATION = "location"
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Taps Aff'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_LOCATION): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the taps aff sensor."""
+    name = config.get(CONF_NAME)
+    location = config.get(CONF_LOCATION)
+
+    taps_aff_data = TapsAffData(location)
```

Comment:
I guess that there will be an exception if the location is not available. For users it will be helpful if there is a log entry that indicate this. Make the setup return if the location is not present this will ensure that the users are not having a non-functional platform for their instance.